### ISSUE TITLE
Explicit literal indentation on top level literal scalars

### ISSUE THREAD

**Issue by ingydotnet** [eemeli/yaml#164]
The YAML 1.2 spec rationalizes indentation level as starting at -1, thus making the first indentation level be 0. This is clever in some respects, but causes a few unfortunate side effects. One of these is the numerical count of top level literal scalars:

The numerical count is supposed to be the number of columns that things are offset. So this,
```
--- |2
    Dear John,
  blah
  blah
```
Should mean `"  Dear John,\nblah\nblah\n"`, and that by looking at it is exactly what you expect.
But if you follow the -1 starting rule then you get `"   Dear John,\n blah\n blah"`.

This will be cleared up in the next version of the YAML spec.

For now, luckily almost every YAML implementation in the wild interprets the first way above.

Your module would do well to do the same.

A related request is that currently your stringifier does this:
```
$ node -e 'console.log(require("yaml").stringify("  foo\nbar\nbaz\n"))'
|1
  foo
bar
baz
```

It would be preferable to do this:
```
$ node -e 'console.log(require("yaml").stringify("  foo\nbar\nbaz\n"))'
--- |2
    foo
  bar
  baz
```

First the document start `---` is really useful to let people know that a top level scalar is actually YAML encoded.
```
--- 42
```
vs
```
42
```

Second is problematic for literal content to be in column 1. A major design factor for literals is that they be able to encode any YAML stream simply by indenting it.

```
---
foo:
  bar: baz
```
could be encoded in another YAML file:
```
---
yaml: |
  ---
  foo:
    bar: baz
```
but without indentation you get:
```
---
yaml: |
---
foo:
  bar: baz
```
which has a completely different meaning.

**Comment by eemeli** [comment]
I think changing the meaning of `|2` for top-level scalars would need to be a semver-major breaking change. It's unfortunate that the 1.2 spec technically requires the current non-obvious implementation. To be clear, I'm not opposed to a new major version, there's a bunch of stuff that's been waiting for it to happen, including a number of deprecated endpoints.

Same probably goes for the explicit document-start/directives-end indicator as a default for scalars; as you mention it changes the output of `YAML.stringify(42)` from `'42\n'` to `'--- 42\n'` or `'---\n  42\n'`, which is rather different.

Indentation of top-level scalars that include lines starting with `---` or `...` will be fixed by #162, but that'll only indent them if required, rather than always. Your yaml-as-literal example has always worked as expected with this library; the upcoming fix is that it'll also work if an explicit document is included as a top-level literal string.

**Comment by eemeli** [comment]
I'm going to close this, because clearly nothing has happened here for over a year. Might revisit this later, if we e.g. end up including a recommendation of some sort about top-level scalars in a later YAML spec version.
